File: src/java/lang/reflect/Method.java
public final class java.lang.reflect.Method extends java.lang.reflect.AbstractMethod
{
    // Constructors
    public final java.lang.reflect.Method();
    // Methods
    private native java.lang.annotation.Annotation getAnnotationNative(java.lang.Class);
    private native [[Ljava.lang.annotation.Annotation; getParameterAnnotationsNative();
    boolean equalNameAndParameters(java.lang.reflect.Method);
    public boolean equals(java.lang.Object);
    public java.lang.annotation.Annotation getAnnotation(java.lang.Class);
    public java.lang.Class getDeclaringClass();
    public native java.lang.Object getDefaultValue();
    public native [Ljava.lang.Class; getExceptionTypes();
    public [Ljava.lang.reflect.Type; getGenericExceptionTypes();
    public [Ljava.lang.reflect.Type; getGenericParameterTypes();
    public java.lang.reflect.Type getGenericReturnType();
    public int getModifiers();
    public java.lang.String getName();
    public [[Ljava.lang.annotation.Annotation; getParameterAnnotations();
    public [Ljava.lang.Class; getParameterTypes();
    public java.lang.Class getReturnType();
    java.lang.String getSignature();
    public [Ljava.lang.reflect.TypeVariable; getTypeParameters();
    public int hashCode();
    public transient native java.lang.Object invoke(java.lang.Object, [Ljava.lang.Object;);
    public boolean isBridge();
    public boolean isDefault();
    public boolean isSynthetic();
    public boolean isVarArgs();
    public java.lang.String toGenericString();
    public java.lang.String toString();
    // Fields
    public static final java.util.Comparator ORDER_BY_SIGNATURE;
}

File: src/java/lang/reflect/Method.ts
class java.lang.reflect.Method extends java.lang.reflect.AbstractMethod {
// Constructors
public constructor() {
return new Packages.java.lang.reflect.Method()
}
// Methods
private native getAnnotationNative(cla: Class): Annotation {
return this.getAnnotationNative(cla)
}
private native getParameterAnnotationsNative(): Annotation[] {
return this.getParameterAnnotationsNative()
}
equalNameAndParameters(met: Method): boolean {
return this.equalNameAndParameters(met)
}
public equals(obj: Object): boolean {
return this.equals(obj)
}
public getAnnotation(cla: Class): Annotation {
return this.getAnnotation(cla)
}
public getDeclaringClass(): Class {
return this.getDeclaringClass()
}
public native getDefaultValue(): Object {
return this.getDefaultValue()
}
public native getExceptionTypes(): Class[] {
return this.getExceptionTypes()
}
public getGenericExceptionTypes(): Type[] {
return this.getGenericExceptionTypes()
}
public getGenericParameterTypes(): Type[] {
return this.getGenericParameterTypes()
}
public getGenericReturnType(): Type {
return this.getGenericReturnType()
}
public getModifiers(): number {
return this.getModifiers()
}
public getName(): string {
return this.getName()
}
public getParameterAnnotations(): Annotation[] {
return this.getParameterAnnotations()
}
public getParameterTypes(): Class[] {
return this.getParameterTypes()
}
public getReturnType(): Class {
return this.getReturnType()
}
getSignature(): string {
return this.getSignature()
}
public getTypeParameters(): TypeVariable[] {
return this.getTypeParameters()
}
public hashCode(): number {
return this.hashCode()
}
public transient native invoke(obj: Object, obj_1: Object[]): Object {
return this.invoke(obj, obj_1)
}
public isBridge(): boolean {
return this.isBridge()
}
public isDefault(): boolean {
return this.isDefault()
}
public isSynthetic(): boolean {
return this.isSynthetic()
}
public isVarArgs(): boolean {
return this.isVarArgs()
}
public toGenericString(): string {
return this.toGenericString()
}
public toString(): string {
return this.toString()
}
// Fields
public static ORDER_BY_SIGNATURE: Comparator
}